src/reactor.cc

```diff
@@ -0,0 +1,343 @@
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  fuel_startup_policy
+      .Init(this, &tritium_storage, std::string("Tritium Storage"),
+            &fuel_tracker, std::string("ss"),
+            startup_inventory,
+            startup_inventory)
+      .Set(fuel_incommod)
+      .Start();
+  blanket_fill_policy
+      .Init(this, &blanket, std::string("Blanket Startup"), &blanket_tracker,
+            std::string("ss"), blanket_size, blanket_size)
+      .Set(blanket_incommod)
+      .Start();
+
+  // Tritium Buy Policy Selection:
+  if (refuel_mode == "schedule") {
+    cyclus::IntDistribution::Ptr active_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(1));
+    cyclus::IntDistribution::Ptr dormant_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(buy_frequency - 1));
+    cyclus::DoubleDistribution::Ptr size_dist =
+        cyclus::FixedDoubleDist::Ptr(new cyclus::FixedDoubleDist(1));
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dist)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              std::string("ss"), reserve_inventory, reserve_inventory)
+        .Set(fuel_incommod);
+  } else {
+    throw cyclus::KeyError("Refill mode " + refuel_mode +
+                           " not recognized! Try 'schedule' or 'fill'.");
+    RecordOperationalInfo("Transaction Error",
+                          "Refill mode " + refuel_mode +
+                              " not recognized! Try 'schedule' or 'fill'.");
+  }
+
+  tritium_sell_policy
+      .Init(this, &tritium_excess, std::string("Excess Tritium"))
+      .Set(fuel_incommod)
+      .Start();
+  helium_sell_policy.Init(this, &helium_storage, std::string("Helium-3"))
+      .Set(he3_outcommod)
+      .Start();
+}
+
+void Reactor::SequesterTritium(){
+  if (sequestered_tritium->quantity() == 0.0){
+    sequestered_tritium = tritium_storage.Pop(sequestered_equilibrium);
+  } else {
+    cyclus::toolkit::MatQuery mq(sequestered_tritium);
+    double equilibrium_deficit = std::max(sequestered_equilibrium - 
+                                          mq.mass(tritium_id), 0.0);
+    sequestered_tritium->Absorb(tritium_storage.Pop(equilibrium_deficit));
+  }
+}
+
+void Reactor::Startup() {
+  cyclus::Material::Ptr initial_storage = tritium_storage.Peek();
+  cyclus::CompMap c = initial_storage->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+
+  if (tritium_storage.quantity() < startup_inventory){
+    throw cyclus::ValueError(
```

Comment:
Do you actually want this to throw a ValueError and break the simulation, or just not start up at this point?